Assets/BossRoom/Scripts/Shared/Game/Entity/NetworkCharacterState.cs

```diff
@@ -13,6 +13,22 @@ public enum LifeState
         Dead,
     }
 
+    /// <summary>
+    /// Describes how the character's movement should be animated: as standing idle, running normally,
+    /// magically slowed, sped up, etc. (Not all statuses are currently used by game content,
+    /// but they are set up to be displayed correctly for future use.)
+    /// </summary>
+    [Serializable]
+    public enum MovementStatus
```

Comment:
As others mentioned above. this enum would be a very good candidate for SO-based/data-driven design.

Reply:
ok (see impl notes above)

Reply:
I've seen the implementation notes above, but they still did not solve having an enum problem I mentioned above. But I am fine whit the changes you did.

Reply:
I'm sorry, I don't understand what change you're suggesting? I don't see a way to remove the enum since it's how the server conveys gameplay status conditions to the client.

Reply:
I meant having a SO-based architecture for enums as well, which makes that code extendable, and you won't need to modify the code base in X different places to add a new MovementStatus.

Reply:
>  I don't see a way to remove the enum since it's how the server conveys gameplay status conditions to the client.

One way that will support what I mentioned above is to have a byte/uint8 representing an ID of the movement status that belongs to the SO, so that later using that ID you can map your request to a specific MovementStatus SO

Reply:
Oh, I see what you mean. I don't think we can do that meaningfully here because this enum is mostly a stub: it represents functionality that didn't make it into the demo yet (like the Slowed condition). I can't make those data-driven yet because I don't know their data: the slowdown amount, the duration, priority over other status conditions (if you're both Slowed and Hasted, which wins?) etc.

And when implementing these, we'd still need to divorce the server effects from the client representation. So we're talking about two new types of SO (server-side data and client-side data) that would basically do nothing right now. That's a lot of boilerplate code for no effect. 

I think we should stick with the enum as a representational implementation just because it's significantly less code. But it would need replacement in a larger fuller game.